### PR TITLE
Support offices page

### PR DESCRIPTION
**Pull request by Mahmoud-Emad**
### Description

This feat is a part of https://github.com/codescalers/CSHR/issues/433#issuecomment-2213415297 

### Changes

- Reorder the SideBar
- Created a new component to serve 404 requests
- Created a new component to serve the office view
- Request to load the office holidays based on the selected day
- Implement a new endpoint to list office holidays
- Support pagination to list paginated offices

### Related Issues

- https://github.com/codescalers/CSHR/issues/441

### Screenshots

- image
- image

### Checklist

- [ ] Tests included
- [ ] Build pass
- [ ] Documentation
- [ ] Code format and docstrings
- [ ] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

File: client/src/components/dialogs/OfficeHolidaysDialog.vue [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+            v-model="requestYears"
+            variant="underlined"
+            label="Last years"
+            color="primary"
+            :rules="[
+              (v: string) => !isNaN(+v) || 'Please enter only numeric values in this field.',
+              (v: string) => +v <= 7 || 'The value should not be more than 7.',
+              (v: string) => +v >= 1 || 'The value should not be less than 1.'
+            ]"
+          /> -->
```
Comment (zaelgohary) [inline client/src/components/dialogs/OfficeHolidaysDialog.vue:26]: Please remove the commented code.

File: server/cshr/serializers/public_holidays.py [hunk 2]
```diff
@@ -16,3 +16,14 @@ class Meta:
...
+
+class OfficePublicHolidaySerializer(ModelSerializer):
+    """
+    This class will be used to get all info about office public holidays.
+    """
+
+
+    class Meta:
+        model = PublicHoliday
+        fields = ["holiday_date", ]
```
Comment (zaelgohary) [inline server/cshr/serializers/public_holidays.py:28]:
```suggestion
        fields = ["holiday_date"]
```

**Comment by zaelgohary** [comment]
I navigated to wrong route and 404 page did not show. 404 route should be added to the router.

image

**Comment by zaelgohary** [comment]
We started using CSHR in 2024 and have no data prior to this year. Do we still need to show empty data for the past four years? I suggest removing them.

I also don't think we need the line `Filtering the total public holidays based on the year 2024.` as it's self-explanatory that filtering will be done based on the chosen year.

Maybe we can add a margin left to `Egypt Office` to be aligned with the rest of the content?

`public holidays` needs to be capitalized as well.

image

File: server/cshr/serializers/public_holidays.py (hunk 2 above)
Comment (mahmoud-emad) [reply server/cshr/serializers/public_holidays.py:28]: This shouldn't be done. If I applied your suggestion, the Python interpreter will read it as a `string` type, while it should be a `tuple` type.

File: client/src/components/dialogs/OfficeHolidaysDialog.vue (hunk 1 above)
Comment (mahmoud-emad) [reply client/src/components/dialogs/OfficeHolidaysDialog.vue:26]: This feature is hidden, but we may need it in the future.

**Comment by Mahmoud-Emad** [comment]
> I navigated to wrong route and 404 page did not show. 404 route should be added to the router.
> 
> image

Please open an issue to support this

**Comment by Mahmoud-Emad** [comment]
> We started using CSHR in 2024 and have no data prior to this year. Do we still need to show empty data for the past four years? I suggest removing them.
> 
> I also don't think we need the line `Filtering the total public holidays based on the year 2024.` as it's self-explanatory that filtering will be done based on the chosen year.
> 
> Maybe we can add a margin left to `Egypt Office` to be aligned with the rest of the content?
> 
> `public holidays` needs to be capitalized as well.
> 
> image

1. The idea about the first point is something like future planning, now we return to the last 5 years, yeah we don't have data now but think about it what will happen next year? and the year after it as well and so on.

2. Removed

3. Removed the card title since we have an alert with the needed info, so it doesn't matter to have the card title component

4. Done

**Review by zaelgohary — APPROVED** [review]

**Pull request merged**